Assignment5/carlomagnani_21A285/Assignment5_CarloMagnani_final.py

```diff
@@ -0,0 +1,297 @@
+ck = ConfigureKGTK(files)
+ck.configure_kgtk(input_graph_path=input_path,
+                  output_path=output_path,
+                  project_name=project_name)
+
+
+# The KGTK setup command defines environment variables for all the files so that you can reuse the Jupyter notebook when you install it on your local machine.
+
+ck.print_env_variables()
+
+get_ipython().run_cell_magic('time', '', 'ck.load_files_into_cache()')
+
+
+# # About this assignment.
+# This assignment is based on https://github.com/usc-isi-i2/kgtk-notebooks/tree/main/tutorial. If you have any questions or doubts, it is encouraged to look how the tutorial performs the different operations.
+# 
+# Additional information can be found in https://kgtk.readthedocs.io/
+
+# ## Simple graph statistics
+# 
+# Let's calculate first some statistics about the KG. Count the number of instances:
+# 
+
+
+kgtk(""" 
+    query -i all
+        --match \'(p)-[r]->(n)\'\
+        --return \'count(p) as instances_num\'
+""")
+
+
+# Now, count the number of distinct properties: 
+# 
+
+
+kgtk("""
+    query -i all
+    --match \'(p)-[r]->(n)\'
+    --return \'count(distinct r.label) as properties\'
+""")
+
+
+# Now, let's count the frequency of those properties. That is, how many instances we can find with each property
+
+
+kgtk("""
+     query -i all
+    --match \'(p)-[r]->(n)\' 
+    --return \'r.label, count(n) as frequency\'
+""")
+
+
+# ## Simple queries
+# Some of these queries are simple and will run in the Wikidata endpoint. 
+# Try both of them using SPARQL and Kypher
+
+
+# Which actors has Schwarzenegger worked with throughout his career? (Print also the movie)
+
+# (in SPARQL)
+
+#select distinct ?movie ?actors{
+#  ?movie wdt:P161 wd:Q2685.
+#  ?movie wdt:P161 ?actors
+#}LIMIT 100
+
+# In Kypher:
+kgtk("""
+    query -i all \
+    --match '(a)<-[:P161]-(f)-[:P161]->(:Q2685)'
+    --return 'f as film, a as actor'
+    / add-labels
+""")
+
+
+# How many awards does Schwarzenegger have?
+
+# SPARQL:
+
+#select distinct ?award{
+#  wd:Q2685 wdt:P166 ?award
+#}LIMIT 100
+
+# Kypher:
+kgtk("""
+    query -i all \
+    --match '(:Q2685)-[:P166]->(awards)' \
+    --return 'count(awards) as Number_of_awards'
+""")
+
+
+# Retrieve at least two members of Schwarzenegger's political party. Make sure only persons are returned
+# SPARQL:
+
+#select ?politician 
+#where{
+#    wd:Q2685 wdt:P102 ?party.
+#    ?politician wdt:P102 ?party.
+#    ?politician wdt:P31  wd:Q5      
+#}LIMIT 2
+
+# Kypher:
+kgtk("""
+    query -i all \
+    --match '(:Q2685)-[:P102]->(party)<-[:P102]-(politician)-[:P31]->(:Q5)' \
+    --return 'politician as politician'
+    --limit 2
+    / add-labels
+""")
+
+
+# What are the properties that describe an artist?
+
+# In theory this one is heavy on Wikidata
+
+# SPARQL: 
+
+#select distinct ?prop 
+#where{
+# ?artist wdt:P106 wd:Q483501.
+# ?artist ?prop ?values
+#}
+
+
+# In Kypher:
+kgtk("""
+    query -i all \
+    --match '(:Q483501)<-[:P31]-(artist)-[p]->()' \
```

Comment:
Not correct. In Wikidata this is done through P106, similarly to what you have done in SPARQL